control_protocol.md

```diff
@@ -0,0 +1,500 @@
+Response.
+
+#### Presentation Display Availablity Request
+
+This message is sent by a presentation controller to find whether a presentation
+display is compatible with potential presentation URLs.  The request may be sent
+when a controller is
+[monitoring the list of available presentation displays](https://w3c.github.io/presentation-api/#dfn-monitor-the-list-of-available-presentation-displays)
+(when starting a presentation or monitoring availability in the background), or
+when a new presentation display is discovered.
+
+```
+Flavor:  Request
+Type:    0x0001
+Subtype: 0x0001
+
+Byte Offset
+  32           +-----------------------+
+               +  NUM_URLS             +
+  34           +-----------------------+
+               +  URL_1_LENGTH         +
+  38           +-----------------------+
+               +  URL_1_CONTENT        +
+               +-----------------------+
+               +  URL_2_LENGTH         +
+               +-----------------------+
+               +  URL_2_CONTENT        +
+               +-----------------------+
+               .                       .
+               +-----------------------+
+               +  URL_N_LENGTH         +
+               +-----------------------+
+               +  URL_N_CONTENT        +
+               +-----------------------+
+```
+
+
+- `NUM_URLS` is an unsigned positive 16-bit integer with the number of URLs
+  contained in the message.
+- `URL_N_LENGTH` is an unsigned 32-bit integer with the length, in bytes, of the
+  Nth URL.
+- `URL_N_CONTENT` is the Nth URL contained in the message, encoded according to
+  [RFC 3986](https://tools.ietf.org/html/rfc3986#section-2).
+  
+Note that this request will send presentation URLs from controllers to
+presentation displays, even before the user has selected a display for
+presentation.  An alternative mechanism would allow the controller to retrieve
+URL patterns from a display that match the URLs supported by that display.
+
+**TODO**: Add messages for this based on conclusion to [Issue #21](issues/21).
+
+#### Presentation Display Availablity Response
+
+The presentation display shall send a response for each Presentation Display
+Availability Request as follows.
+
+```
+Flavor:  Response
+Type:    0x0001
+Subtype: 0x0002
+
+Byte Offset
+  40           +-----------------------+
+               + NUM_URLS              +
+  42           +-----------------------+
+               + AVAILABILITY_RESULT_1 +
+  43           +-----------------------+
+               .                       .
+               +-----------------------+
+               + AVAILABILITY_RESULT_N +
+               +-----------------------+
+```
+
+- `NUM_URLS` is an unsigned positive 16-bit integer that is identical to
+  `NUM_URLS` in the corresponding Presentation Display Availability Request.
+- Each `AVAILABILITY_RESULT` is an unsigned 8-bit integer containing the
+  availability result for the Nth URL in the request.  The results are as follows:
```

Comment:
Receiver side might have its own preference of the compatible URLs. We can extend this message by sending a list of (index, result) tuples.

Reply:
Good point, however, the current algorithm for starting presentation allows the controller to determine the URL priority:

https://w3c.github.io/presentation-api/#dfn-start-a-presentation-connection, step 5.

Meaning, if the page creates PresentationReuqest([URL1, URL2]), and the receiver reports [URL2, URL1] as compatible, the page must still initiate presentation with URL1.

However, I could see an argument for allowing the receiver to influence this priority, for example preferring pre-installed apps for some URLs.  So I'll add this enhancement in case we update the controller algorithms get updated.